modules/ztp-image-based-upgrade-generate-seed-image.adoc

```diff
@@ -0,0 +1,221 @@
+
+. Detach the seed cluster manually or provide the `hubKubeConfig` information in the `Secret` CR.
+
+.. (Optional) To manually detach the seed cluster from the hub cluster, run the following command:
++
+[source,terminal]
+----
+$ oc delete managedcluster sno-worker-example
+----
+
+.. Wait until the `ManagedCluster` CR is removed. Once the CR is removed, create the proper `SeedGenerator` CR. The {lcao} cleans up the {rh-rhacm} artifacts.
+
+. Create the `Secret` CR in the `openshift-lifecycle-agent` namespace.
++
+--
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: seedgen <1>
+  namespace: openshift-lifecycle-agent
+type: Opaque
+data:
+  seedAuth: <encoded_AUTHFILE> <2>
+  hubKubeconfig: <encoded_KUBECONFIG> <3>
+----
+<1> The `Secret` CR must have the `name: seedgen` and `namespace: openshift-lifecycle-agent` fields.
+<2> Specifies a base64-encoded authfile for write-access to the registry for pushing the generated seed images.
+<3> Optional. Specifies a base64-encoded `KUBECONFIG` file for administrator access to the hub cluster to detach the seed cluster from {rh-rhacm}. If this is not present in the secret, the {rh-rhacm} cleanup is skipped and the seed cluster must be detached from the hub cluster manually.
+--
+
+.. Create the authentication file by running the following command:
++
+--
+.Authentication file
+[source,terminal]
+----
+$ MY_USER=myuserid
+$ AUTHFILE=/tmp/my-auth.json
+$ podman login --authfile ${AUTHFILE} -u ${MY_USER} quay.io/${MY_USER}
+----
+
+[source,terminal]
+----
+$ base64 -w 0 ${AUTHFILE} ; echo
+----
+--
+
+.. (Optional) Create the `Kubeconfig` file by running the following command:
++
+--
+.hubKubeconfig file
+[source,terminal]
+----
+$ HUBKUBECONFIGFILE=/home/<user_ID>/hubkubeconfig
+----
+
+[source,terminal]
+----
+$ base64 -w 0 ${HUBKUBECONFIGFILE} ; echo
+----
+--
+
+.. Copy the outputs into the respective `seedAuth` and `hubKubeconfig` fields in the `Secret` CR.
+
+. Apply the `Secret` CR.
++
+[source,terminal]
+----
+$ oc apply -f secretseedgenerator.yaml
+----
+
+. Create the `SeedGenerator` CR:
++
+--
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: SeedGenerator
+metadata:
+  name: seedimage <1>
+spec:
+  seedImage: <seed_container_image> <2>
+----
+<1> The `SeedGenerator` CR must be named `seedimage`.
+<2> Specify the container image URL, for example, `quay.io/example/seed-container-image:<tag>`. It is recommended to use the `<seed_cluster_name>:<ocp_version>` format.
+--
+
+. Generate the seed image by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f seedgenerator.yaml
+----
+
++
+[IMPORTANT]
+====
+The cluster reboots and loses API capabilities while the {lcao} generates the seed image.
+Applying the `SeedGenerator` CR stops the `kubelet` and the CRI-O operations, then it starts the image generation.
+====
+
+Once the image generation is complete, the cluster is reattached to the hub cluster and can be accessed through the API.
```

Comment:
does this happen automatically even if someone didn't supply the optional hub kubeconfig, or would they have to manually add it back in to the cluster?

Reply:
It does not happen automatically, I just revised this part to make it obvious it's a manual reattachment.

I also clarified with the team that the hub kubeconfig option is not customer facing, so any mention of that has been removed, ([see here](https://69809--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-talm-updating-managed-policies#ztp-image-based-upgrade-seed-generation_ztp-talm))